### PR TITLE
How to design infinite impulse response filters using Matlab

### LINKED ISSUE #2848: How to Design Infinite Implulse Response (IIR) Filters using Matlab
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Introduction:
IIR filters are digital filters with infinite pulse response and have a feedback. They have a better response and their phase characteristics are non- linear which can cause a problem to the system which need phase linearity. This is why they are not preferred for use in digital signal processing when the phase is of essence.
The design of IIR filters is not straight forward as that of FIR filters, however Matlab signal processing toolbox provides a number of advanced routines to assist in this process. In this article, we will look how to use Matlab to design an IIR filters.

### Key takeaways: 
- Theory behind filters.
- Applications of IIR filters.
- Matlab's program for designing the filters.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by atienodorine3**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

this fixes #2848

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your topic suggestion.
Your topic form has been entered into our queue and should be reviewed (for approval) as soon as a content moderator is finished reviewing the ones in the queue before it.  @atienodorine3

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. :rocket:
Thank you.

File: content/articles/how-to-design-infinite-impulse-response-filters-using-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+### HOW TO DESIGN AN INFINITE IMPULSE RESPONSE FILTERS(IIR)
+### Introduction
+To filter is to remove the unwanted properties of a signal. IIR filters with an infinite number of impulses. Here the output y(n) response depends on the present input x(n), previous input x(n-1) as well as the previous output y(n-1). The differential equation for the IIR filter can be given by the differential equation $y(n)=b_0x(n)+b_1x(n-1)+---b_mx(n-m)-a_1y(n-1)---a_ny(m-n)$. To see more about this, you can get it [here](https://www.sciencedirect.com/topics/engineering/iir-filters). Since the filter uses the previous outputs, there is feedback from the filter structure. The design of an IIR filter is based on the transfer function($H_s$) that satisfies the requirement of the filter requirement. The transfer function($H_s$) is the ratio of the output response to the input response.
```
Comment (jumamiller) [inline content/articles/how-to-design-infinite-impulse-response-filters-using-matlab/index.md:3]: A great intro here, but could you twist it a little: by adding the following:    

-  Tell us what's an IIR in detail,for example`The infinite impulse response (IIR) filter is a recursive filter in that the output from the filter is computed by using the current and previous inputs and previous outputs`.  This has more meaning than the one you have provided!

**Review by jumamiller — APPROVED** [review]
@atienodorine3 , a great article. Looking forward for more  from you!

**Comment by hectorkambow** [comment]
@atienodorine3 

I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[How to design infinite impulse response filters using Matlab #3022.pdf](https://github.com/section-engineering-education/engineering-education/files/6963271/How.to.design.infinite.impulse.response.filters.using.Matlab.3022.pdf)

**Comment by hectorkambow** [comment]
@atienodorine3 
Also upon a quick glance the tutorial/article seems rather short. 

cc @jumamiller

**Comment by atienodorine3** [comment]
@hectorkambow I will enhance my article and add more content
cc @jumamiller  and check on the plagarism.

**Comment by atienodorine3** [comment]
@hectorkambow requested changes done.
cc @jumamiller

**Review by jumamiller — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@atienodorine3 

Please be sure to go over any edits/revisions made by your peer reviewers or myself to improve future article 👍 

cc @jumamiller 
Please do the same made some major edits via my final - please review to better improve future reviews.

**Pull request merged**